src/io/iter_sampler.cc

```diff
@@ -126,7 +126,9 @@ class RandomSampler : public IIterator<DataInst> {
     param_.InitAllowUnknown(kwargs);
     indices_.resize(param_.length);
     std::iota(std::begin(indices_), std::end(indices_), 0);  // fill like arange
-    rng_.reset(new common::RANDOM_ENGINE(kRandMagic + param_.seed));
+    mshadow::Random<cpu> *ctx_rng = ResourceManager::Get()->Request(
+      Context::CPU(), ResourceRequest::kRandom).get_random<cpu, real_t>(nullptr);
+    rng_.reset(new common::RANDOM_ENGINE(ctx_rng->GetSeed() + param_.seed));
```

Comment:
Delete the user-facing `seed` parameter now that you're using the MXNet seed?

Reply:
Removed